### PR TITLE
Breakout startup wait to new file & update Batch test

### PR DESCRIPTION
**Pull request by nick-stroud**
Breakout the common Ansible task of waiting for startup script to finish so that it can be reused in various tests.

### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [ ] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [x] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

**Comment by nick-stroud** [comment]
@tpdownes & @heyealex, I am looking for initial feedback on this. I waited to apply this in more places until we agree it is a good idea. If we like this pattern then I will go ahead and swap out this task anywhere we are waiting for startup script.

File: tools/cloud-build/daily-tests/ansible_playbooks/spack.yml
Comment (nick-stroud) [inline tools/cloud-build/daily-tests/ansible_playbooks/spack.yml:23]: According to [ansible docs](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/wait_for_module.html#parameter-state), when used for a file/search string (which is what we are doing), `present` has the same effect as the default `started`.

File: tools/cloud-build/daily-tests/ansible_playbooks/spack.yml
Comment (heyealex) [inline tools/cloud-build/daily-tests/ansible_playbooks/spack.yml:17]: I'm curious if you could do the same thing with a role, similar to this: https://github.com/dellhpc/omnia/blob/v1.2.2/omnia.yml#L20

I think you just need to create a "roles" directory with that name in it, then you can run multiple roles with one task. I wouldn't put much time into it, but if it's easy it could be worth it and decrease the boilerplate.

File: tools/cloud-build/daily-tests/ansible_playbooks/spack.yml
Comment (tpdownes) [reply tools/cloud-build/daily-tests/ansible_playbooks/spack.yml:23]: I agree with the logic. I'd raise at a "nit" level that I prefer being explicit for state for clarity and for possibility of default behavior changes. This is probably not one of those cases.

**Review by tpdownes — APPROVED** [review]
I like the idea here and believe it's a step in right direction.

@nick-stroud I prefer you wait until #435 is merged

File: tools/cloud-build/daily-tests/ansible_playbooks/spack.yml
Comment (nick-stroud) [reply tools/cloud-build/daily-tests/ansible_playbooks/spack.yml:23]: Ack.

**Comment by nick-stroud** [comment]
This had to be rebased past ansible-lint upgrade and needs re-approval. Integration tests previously passed but I kicked them off again anyway.

**Review by tpdownes — APPROVED** [review]

**Pull request merged**
